README.md

```diff
@@ -1,4 +1,87 @@
-eiconv [![Build Status](https://secure.travis-ci.org/mmzeeman/eiconv.png?branch=master)](http://travis-ci.org/mmzeeman/eiconv)
-======
+# Erlang NIF wrapper for iconv
 
-Erlang NIF wrapper for iconv
+[![Build Status](https://secure.travis-ci.org/zotonic/eiconv.png?branch=master)](http://travis-ci.org/zotonic/eiconv)
+
+An Erlang wrapper for the character set conversion utility [iconv](https://en.wikipedia.org/wiki/Iconv).
+
+## Getting started
+
+### Include as a dependency
+
+Using [rebar3](http://www.rebar3.org/), add the dependency to your `rebar.config` file:
+
+```erlang
+{deps, [
+	{eiconv, ".*", {git, "git://github.com/zotonic/eiconv.git", {branch, "master"}}}
+]}.
+```
+
+You must also enable the compilation of `eiconv_nif.so` through [port_compiler](https://github.com/blt/port_compiler). Again, add in your `rebar.config` file:
+
+```erlang
+{overrides,
+ [{override, eiconv,
+   [
+	{plugins, [pc]},
+	{port_env, [{"darwin|freebsd|openbsd", "LDFLAGS", "$LDFLAGS -liconv"},
+				{"freebsd|openbsd", "CFLAGS", "$CFLAGS -I/usr/local/include"},
+				{"freebsd|openbsd", "LDFLAGS", "$LDFLAGS -L/usr/local/lib"}]},
+
+	{port_specs, [{"priv/eiconv_nif.so", ["c_src/*.c"]}]},
+	{artifacts, ["priv/eiconv_nif.so"]},
```

Comment:
Is this still necessary with rebar3? In Zotonic we don’t have this (only in gen_smtp) and everything works fine.

Reply:
There is no other build process whatsoever that handle that part?

I've tested again on https://github.com/YtoTech/geo-sensors-gateway, without this port_compiler configuration the `priv.eiconv_nif.so` is not built.

I'm new to that so I'm not sure I do that well.